nltk/tokenize/casual.py

```diff
@@ -299,15 +341,40 @@ def tokenize(self, text):
             text = reduce_lengthening(text)
         # Shorten problematic sequences of characters
         safe_text = HANG_RE.sub(r"\1\1\1", text)
-        # Tokenize:
-        words = WORD_RE.findall(safe_text)
+        # Recognise phone numbers during tokenization
+        if self.match_phone_numbers:
+            words = self.PHONE_WORD_RE.findall(safe_text)
+        else:
+            words = self.WORD_RE.findall(safe_text)
         # Possibly alter the case, but avoid changing emoticons like :D into :d:
         if not self.preserve_case:
             words = list(
                 map((lambda x: x if EMOTICON_RE.search(x) else x.lower()), words)
             )
         return words
 
+    @property
+    def WORD_RE(self) -> regex.Pattern:
+        """Core TweetTokenizer regex"""
+        # Compiles the regex for this and all future instantiations of TweetTokenizer.
+        if not type(self)._WORD_RE:
+            type(self)._WORD_RE = regex.compile(
+                f"({'|'.join(REGEXPS)})",
+                regex.VERBOSE | regex.I | regex.UNICODE,
+            )
+        return type(self)._WORD_RE
+
+    @property
+    def PHONE_WORD_RE(self) -> regex.Pattern:
```

Comment:
naming nit: maybe `PHONE_NUM_RE` since it's a phone number?

Reply:
I'm not sold on this - `PHONE_NUM_RE` sounds like it's *just* a regex for phone numbers, while it's actually `WORD_RE` *plus* phone numbers. That said, `PHONE_WORD_RE` isn't a great name.